File: 백준/Silver/10828. 스택/스택.py
n = int(input())    # 명령어의 수
dir = []            # 명령어를 담을 배열열
answer = []
for _ in range(n):
    instruction = list(map(str, input().split()))
    if instruction[0] == "push":
        num = int(instruction[1])
        dir.append(num)
    elif instruction[0] == "pop":
        if dir == []:
            answer.append(-1)
            continue
        answer.append(dir[-1])
        dir = dir[:-1]
    elif instruction[0] == "size":
        answer.append(len(dir))
    elif instruction[0] == "empty":
        if dir == []:
            answer.append(1)
        else:
            answer.append(0)
    else:
        if dir == []:
            answer.append(-1)
            continue
        answer.append(dir[-1])

for i in answer:
    print(i)   

File: 백준/Silver/10828. 스택/스택.cc
#include <bits/stdc++.h>
using namespace std;


int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    stack<int> s;
    int num;
    string cmd;
    cin >> num;
    while (num--)
    {
        cin >> cmd;
        if (cmd == "push")
        {
            int i;
            cin >> i;
            s.push(i);
        }
        else if (cmd == "pop")
        {
            if (!s.empty())
            {
                cout << s.top() << '\n';
                s.pop();
            }
            else
            {
                cout << -1 << '\n';
            }
        }
        else if (cmd == "size")
        {
            cout << s.size() << '\n';
        }
        else if (cmd == "empty")
        {
            if (!s.empty()) cout << 0 << '\n';
            else cout << 1 << '\n';
        }
        else // "top"
        {
            if (!s.empty())
            {
                cout << s.top() << '\n';
            }
            else
            {
                cout << -1 << '\n';
            }
        }
    }
}
